.github/workflows/test.yml

```diff
@@ -30,11 +30,14 @@ jobs:
           enable-cache: true
           python-version: ${{ matrix.python-version }}
           version: ${{ env.UV_VERSION }}
-      - name: Install dependencies
-        run: uv sync --group plugin-test --frozen
-      - name: Test with pytest
-        run: |
-          uv run pytest -n logical --dist loadgroup --junit-xml=test-results.xml
+      - name: Install mandatory dependencies
+        run: uv sync --frozen --group plugin-test
```

Comment:
I like the idea of this PR. Maybe we should mark the tests that need the plugins in this group as well, so that we can just do a bare sync for the main tests? Maybe even just have this as the only group that gets installed later instead of also 'all'

Reply:
Is there an advantage to having 2 separate marks and groups here? I was thinking we just have one mark, something like `@pytest.mark.needs_optional_deps`
Then we'd add all the optional deps that get tested to the plugin-test group (or perhaps there is a more descriptive name that would be good... optional-deps-for-tests?)
From my perspective the groups that turn into 'extras' aren't really special here, and we don't have tests for most of them anyway.

Reply:
I misunderstood your meaning earlier, and now I’ve marked them with `@pytest.mark.require_optional_deps`.